File: 6kyu 1/persistence.js
// Persistent Bugger.


function persistence(num) {
   let count = 0;
   while (num >= 10) {
      const arr = [];
      const str = num.toString();
      for (let i in str) {
         arr.push(Number(str[i]));
      }
      num = arr.reduce((acc, curr) => acc * curr);
      count++;
   }
   return count;
}


const persistence = (n, c) => n > 9 ? persistence(n.toString().split('').reduce((x, y) => x * y), c ? (c + 1) : 1) : c ? c : 0;


function persistence(n, s = `${n}`, c = 0) {
   while (s.length > 1) n = s.split('').reduce((t, a) => +t * +a), s = `${n}`, c++;
   return c;
}


const persistence = num => {
   return num > 9 ? persistence([...`${num}`].reduce((a, b) => a * +b)) + 1 : 0;
}


console.log(persistence(39), 3);
console.log(persistence(4), 0);
console.log(persistence(25), 2);
console.log(persistence(999), 4);

File: Python/6py 3/persistence.py
"""
|--------------------------------------------------------------------------
| Persistent Bugger.
|--------------------------------------------------------------------------
|
"""

from functools import reduce
def persistence(n):
   count = 0
   while (n >= 10):
      arr = []
      string = str(n)
      for i in string:
         arr.append(int(i))
      n = reduce(lambda a, b: a * b, arr)
      count += 1
   return count


from operator import mul
def persistence(n):
   return 0 if n<10 else persistence(reduce(mul,[int(i) for i in str(n)],1))+1


def persistence(n):
   if n < 10: return 0
   mult = 1
   while(n > 0):
      mult = n%10 * mult
      n = n//10
   return persistence(mult) + 1


def persistence(n, count=0):
   return count if n<10 else persistence( eval('*'.join(list(str(n)))), count+1 )


print(persistence(39), 3)
print(persistence(4), 0)
print(persistence(25), 2)
print(persistence(999), 4)